sys/net/gnrc/network_layer/ipv6/nib/_nib-router.c

```diff
@@ -228,7 +274,9 @@ static void _snd_ra(gnrc_netif_t *netif, const ipv6_addr_t *dst,
 {
     gnrc_pktsnip_t *ext_opts = NULL;
 
-    if (!final) {
+    if (final) {
```

Comment:
I think another check for `IS_ACTIVE(CONFIG_GNRC_IPV6_NIB_RIO_IN_LAST_RA)` is needed here.

Reply:
I deliberately moved it to `_build_final_ext_opts()` to keep this part of the code tidy.